packages/sdk/src/client.ts

```diff
@@ -2503,17 +2507,19 @@ export class Client
                 check(isDefined(expectedHash), 'expected hash not found in error')
                 const expectedMiniblockNum = getRpcErrorProperty(err, 'expNum')
                 check(isDefined(expectedMiniblockNum), 'expected miniblock num not found in error')
-                return await this.makeEventWithHashAndAddToStream(
-                    streamId,
-                    payload,
+                return await makeRetry(
                     bin_fromHexString(expectedHash),
                     BigInt(expectedMiniblockNum),
-                    optional,
-                    isDefined(localId) ? eventId : undefined,
-                    cleartext,
-                    tags,
-                    retryCount + 1,
                 )
+            } else if (
+                // for blockchain transactions: if we get a permission denied error, and the error message contains 'Transaction has 0 confirmations',
+                // we need to retry the event, since the node is likely lagging behind, and waiting a few seconds will allow it to catch up
+                errorContains(err, Err.PERMISSION_DENIED) &&
+                errorContainsMessage(err, 'Transaction has 0 confirmations') &&
```

Comment:
The message ends with a `.` in the node. But this will work.

The node end can also be improved and the client. It currently retrieves the tx receipt, if fails -> error (not converted to RiverErr) and the client won't retry. If the receipt isn't found the error should be mapped to a `Err.NOT_FOUND` and also retried because the client knows that the receipt is available but the client is lagging behind. I can make this change in the node and push it on this branch if you want.